### PR TITLE
change: ETCM-8366 Make native token scripts optional

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

Makes the scripts optional in the native token pallet. If they're absent, the IDP will not query the main chain state at all and will not trigger the inherent.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [x] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: primitives/native-token-management/src/lib.rs [hunk 1]
```diff
@@ -119,9 +119,12 @@ mod inherent_provider {
 			C::Api: NativeTokenManagementApi<Block>,
 		{
```
Comment (kpinter-iohk) [inline primitives/native-token-management/src/lib.rs:120]: Apologies for the stylistic nitpickery but the mid-function `return` bothered me. GH doesn't let me put this in a normal suggestion:
```rust

		pub async fn new<Block, C>(
			client: Arc<C>,
			data_source: &(dyn NativeTokenManagementDataSource + Send + Sync),
			mc_hash: McBlockHash,
			parent_hash: <Block as BlockT>::Hash,
		) -> Result<Self, IDPCreationError>
		where
			Block: BlockT,
			C: HeaderBackend<Block>,
			C: ProvideRuntimeApi<Block> + Send + Sync,
			C::Api: NativeTokenManagementApi<Block>,
		{
			let scripts_opt = client
				.runtime_api()
				.get_main_chain_scripts(parent_hash)
				.map_err(IDPCreationError::GetMainChainScriptsError)?;
			let token_amount = match scripts_opt {
				None => None,
				Some(scripts) => {
					let parent_mc_hash: Option<McBlockHash> =
						get_mc_hash_for_block(client.as_ref(), parent_hash)
							.map_err(IDPCreationError::McHashError)?;
					let token_amount = data_source
						.get_total_native_token_transfer(
							parent_mc_hash,
							mc_hash,
							scripts.native_token_policy_id,
							scripts.native_token_asset_name,
							scripts.illiquid_supply_validator_address,
						)
						.await?;

					Some(token_amount).filter(|amount| amount.0 > 0)
				},
			};
			Ok(Self { token_amount })
		}
```

File: primitives/native-token-management/src/lib.rs [hunk 2]
```diff
@@ -145,10 +150,11 @@ mod inherent_provider {
 			&self,
 			inherent_data: &mut InherentData,
 		) -> Result<(), sp_inherents::Error> {
-			inherent_data.put_data(
-				INHERENT_IDENTIFIER,
-				&TokenTransferData { token_amount: self.token_amount.clone() },
-			)
+			if let Some(token_amount) = self.token_amount.clone() {
```
Comment (kpinter-iohk) [inline primitives/native-token-management/src/lib.rs:153]: Clippy is telling me you don't need the `clone`

**Review by kpinter-iohk — APPROVED** [review]
Left some minor comments but otherwise LGTM.

**Review by LGLO — APPROVED** [review]

File: primitives/native-token-management/src/lib.rs (hunk 1 above)
Comment (AmbientTea) [reply primitives/native-token-management/src/lib.rs:120]: I know this feels wrong coming from haskell, but I'd rather have an early return than nest the code here

Comment (kpinter-iohk) [reply primitives/native-token-management/src/lib.rs:120]: Sure, I'd just like to point out that single return is not a Haskell thing

**Pull request merged**